src/hooks/useEtherspotModules.ts

```diff
@@ -91,7 +101,73 @@ const useEtherspotModules = (chainId?: number): IEtherspotModulesHook => {
     }
   }
 
-  return { installModule, uninstallModule };
+
+  const listModules = async (
+    pageSize?: number,
+    accountAddress?: string,
+    modulesChainId: number = defaultChainId,
+  ) => {
+    // this hook can only be used is the sdk is using the modular functionality
+    if (!isModular) {
+      throw new Error(`The <EtherspotTransactionKit /> component is not using the modular functionality. Please make sure to use the modular functionality to install and uninstall modules.`);
+    }
+
+    const sdkForChainId = await getSdk(modulesChainId) as ModularSdk;
+
+    const modulesForAccount = accountAddress ?? await sdkForChainId.getCounterFactualAddress();
+    if (!modulesForAccount) {
+      throw new Error(`No account address provided!`);
+    }
+
+    // If no pageSize is passed as an arg, the default should be 50
+    const modulesListPageSize = pageSize ?? 50;
+
+    try {
+        const getAllModules = await sdkForChainId.getAllModules(modulesListPageSize);
+        return getAllModules;
+    } catch (e) {
+      console.error(
+        `Sorry, an error occurred whilst trying to list all your wallet modules on`
+        + ` ${modulesForAccount}. Please try again. Error:`,
+        e,
+      );
+      throw new Error(`Failed to list all modules: ${e}`)
```

Comment:
Should be `e.message` otherwise the error may print the whole stack trace

```suggestion
      throw new Error(`Failed to list all modules: ${e.message}`)
```